features/authentication/contracts/social_auth.md

```diff
@@ -0,0 +1,37 @@
+# Social auth
+
+Because we are using firebase as source of truth for our social authentication
+flow, all the resources will be focused with this in mind.
+
+## Social sign in & sign up [/auth/social]
+
+This resource represents a basic social authentication flow.
+
+### Social sign in & sign up [] [POST] 🔓
+
+The client should first use firebase sdk to sign in with the corresponding
+social authentication provider (web or mobile based). Once logged through
+firebase the client app should retrieve the **id_token** and send it to the back-end.
+The backend will be able with the firebase sdk, to authenticate the user and
+get its data.
+
+It's **important** that the back-end has a system implemented to identify if a
+user exists with the given email and if it does associate it to this social
+auth instance; and if not create a new one. This will cover a sign in and sign up
+process through social auth.
+
+* **Body**
+  ```json
+  {
+    "firebase_token": "id_token"
+  }
+  ```
+
```

Comment:
I think we should have in mind that most social networks store partial information compared to the information we use to store in our apps of the user.

For example, in some social networks we store the user gender or some required specific fields for the context of the app.

There should be an state that means the account is not fulfilled. And an endpoint to do it.

Reply:
Yes that's why I thought on basics, mail and name. Rest of profile info I think it should be handled by each app use case. Maybe the app after first login has a full onboarding process in which some info could be extracted from social auth and other info no. That's why I kept it simple.